src/Hosting/TestHost/src/RequestLifetimeFeature.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    internal class RequestLifetimeFeature : IHttpRequestLifetimeFeature
+    {
+        private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
+        public CancellationToken RequestAborted
+        {
+            get => _cancellationTokenSource.Token;
+            set => throw new NotSupportedException();
```

Comment:
You can support with set like this: https://github.com/grpc/grpc-dotnet/blob/0be935fca79674be56c9fe7e69f6708306889ae2/testassets/FunctionalTestsWebsite/Infrastructure/TestHttpRequestLifetimeFeature.cs

Reply:
That's lying to the API user though. That setter should not exist IMO as setting it breaks the Abort functionality.